### ISSUE TITLE
InputScrollFrameTemplate errors OnLoad if instantiated via CreateFrame

### ISSUE THREAD

**Issue by Meorawr** [Stanzilla/WoWUIBugs#300]
Creating a ScrollFrame via the [CreateFrame](https://wowpedia.fandom.com/wiki/API_CreateFrame) API that inherits [InputScrollFrameTemplate](https://github.com/Gethe/wow-ui-source/blob/15a65c6cfd23400d23c012c6ee4f1375e4b16abd/Interface/SharedXML/SecureUIPanelTemplates.xml#L135-L142) will trigger an error originating from a failed call to `self:SetMaxLetters(maxLetters)` inside [InputScrollFrame_OnLoad](https://github.com/Gethe/wow-ui-source/blob/15a65c6cfd23400d23c012c6ee4f1375e4b16abd/Interface/SharedXML/SecureUIPanelTemplates.lua#L344-L357), as the `self.maxLetters` field will be nil upon initialization.

This typically won't affect frames inheriting this template in XML as they have an opportunity to define the necessary key/value pairs statically prior to the OnLoad script firing, however for addons using this template via the API there's no such chance to define the `self.maxLetters` field - which may break a few users. It'd be nice if the call to SetMaxLetters could be skipped if that field is nil.

To reproduce the issue, run the following chat command:

```
/run local f = CreateFrame("ScrollFrame", nil, nil, "InputScrollFrameTemplate"); f:SetPoint("CENTER"); f:SetSize(1, 1);
```

The error report this will generate is as follows:

```
Message: Interface/SharedXML/SecureUIPanelTemplates.lua:357: Usage: self:SetMaxLetters(maxLetters)
Time: Mon Oct 10 12:56:53 2022
Count: 1
Stack: Interface/SharedXML/SecureUIPanelTemplates.lua:357: Usage: self:SetMaxLetters(maxLetters)
[string "=[C]"]: in function `SetMaxLetters'
[string "@Interface/SharedXML/SecureUIPanelTemplates.lua"]:357: in function <Interface/SharedXML/SecureUIPanelTemplates.lua:344>
[string "=[C]"]: in function `CreateFrame'
[string "CreateFrame("ScrollFrame", nil, nil, "InputScrollFrameTemplate")"]:1: in main chunk
[string "=[C]"]: in function `RunScript'
[string "@Interface/FrameXML/ChatFrame.lua"]:2165: in function `?'
[string "@Interface/FrameXML/ChatFrame.lua"]:5219: in function `ChatEdit_ParseText'
[string "@Interface/FrameXML/ChatFrame.lua"]:4883: in function `ChatEdit_SendText'
[string "@Interface/FrameXML/ChatFrame.lua"]:4919: in function `ChatEdit_OnEnterPressed'
[string "*ChatFrame.xml:127_OnEnterPressed"]:1: in function <[string "*ChatFrame.xml:127_OnEnterPressed"]:1>

Locals: (*temporary) = EditBox {
 0 = <userdata>
 Instructions = FontString {
 }
}
(*temporary) = nil
```

**Comment by ls-** [comment]
~~The next line `self.EditBox.Instructions:SetText(self.instructions)` will also error out because `self.instructions` is nil when created via `CreateFrame`. `:SetText` no longer accepts nil inputs either.~~

Bruh...

**Comment by Meorawr** [comment]
It won't error - `FontString:SetText()` continues to accept nil inputs in 10.0. _Other_ SetText APIs such as `EditBox:SetText()`  however don't, because consistency :)

**Comment by Arxareon** [comment]
This is a major issue. Given how close we are to release, I fear it'll be left bugged. Has some kind of workaround been found already? All of my addons are written in LUA using the API, it'd be quite the work to rewrite everything to work with XML..

**Comment by Meorawr** [comment]
As todays' 10.0.0 PTR build had minimal changes I'd lean toward it being unlikely that this'll be solved for the deployment of the prepatch.

There's a chance that any of the builds that'll get deployed to live in the interim between prepatch and 10.0.2 _may_ fix the issue, but it depends on other priorities - there's going to be a _lot_ of other issues competing for attention over the next few weeks, and this issue falls into that awkward gap where it doesn't impact any of Blizzards' usages of the template and instead only applies to addons.

As for workarounds, you've got a few options. A simple one would be to include the following template in your addon and replace all usages of InputScrollFrameTemplate with it - just make sure you replace "MyAddOn" with something unique (like your own addon name) as appropriate.

```xml
<Ui>
	<Frame name="MyAddOn_InputScrollFrameTemplate" virtual="true" inherits="InputScrollFrameTemplate">
		<Scripts>
			<OnLoad inherit="append">
				local SetMaxLetters = self.EditBox.SetMaxLetters;
				self.EditBox.SetMaxLetters = function(editbox, count)
					if count then
						SetMaxLetters(editbox, count);
					end
				end;
			</OnLoad>
		</Scripts>
	</Frame>
</Ui>
```

**Comment by Meorawr** [comment]
This should hopefully be fixed in an upcoming 10.0.5 PTR build.

**Comment by Meorawr** [comment]
Fixed in 10.0.5.47481.
